app/src/main/java/it/ministerodellasalute/immuni/workers/ExposureNotificationWorkers.kt

```diff
@@ -169,6 +169,108 @@ class RequestDiagnosisKeysWorker(
+                if (settingsResult is FetchSettingsResult.Success) {
+                    // cleanup entities that are older than DAYS_OF_SELF_ISOLATION
+                    exposureReportingRepository.deleteOldSummaries(settingsResult.serverDate)
+                } else {
+                    return@withTimeout Result.retry()
+                }
+
+                if (settingsManager.isAppOutdated) {
+                    return@withTimeout Result.retry()
+                }
+
+                val serverDate = settingsResult.serverDate
+
+                analyticsManager.setup(serverDate)
+
+                chunksDir.apply {
+                    deleteRecursively()
+                    mkdir()
+                }
+                val countries = exposureReportingRepository.getCountriesOfInterest()
+                val diagnosisEuKeyList = mutableListOf<File>()
+                countries.forEach { country ->
+                    val indexResponse = immuniApiCall { api.indexEu(country.code) }
+                    if (indexResponse !is NetworkResource.Success) {
+                        val error = indexResponse.error
+                        if (error is NetworkError.HttpError && error.httpCode == 404) {
+                            return@withTimeout success(serverDate)
+                        }
+
+                        return@withTimeout Result.retry()
+                    }
+                    val data = indexResponse.data ?: return@withTimeout Result.retry()
+                    val lastProcessedChunkSuccessor = country.lastProcessedChunk + 1
+                    val currentOldest = max(data.oldest, lastProcessedChunkSuccessor)
+                    val chunkRange = (currentOldest..data.newest).toList().takeLast(20)
+                    for (currentChunk in chunkRange) {
+                        val chunkResponse =
+                            immuniApiCall { api.chunkEu(country.code, currentChunk) }
+                        if (chunkResponse !is NetworkResource.Success) {
+                            return@withTimeout Result.retry()
+                        }
+                        val filePath =
+                            listOf(chunksDirPath, "$currentChunk.zip").joinToString(File.separator)
+                        try {
+                            chunkResponse.data?.byteStream()?.saveToFile(filePath)
+                                ?: return@withTimeout Result.retry()
+                            diagnosisEuKeyList.addAll(listOf(File(filePath)))
+                            country.lastProcessedChunk = data.newest
+                        } catch (e: Exception) {
+                            return@withTimeout Result.retry()
+                        }
+                    }
+                }
+                if(diagnosisEuKeyList.isNotEmpty()){
+                exposureManager.provideDiagnosisKeys(
+                    keyFiles = diagnosisEuKeyList,
```

Comment:
Attention, this will not work! Due to EN SDK constraints you must push one chunk at the time to the `provideDiagnosisKeys`.

Also, since you have to do that, you are more likely to hit the daily limit.